### ISSUE TITLE
odd name for screenshot file

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/joist#286]
Mac OS X 10.10.4 + Chrome latest, but this issue is not OS or browser dependent.

The filename created for a screenshot is of the form `-TITLE- screenshot.png`, where TITLE is the simulation's title. What is the purpose of the hypens surrounding the sim name? 

Examples:

http://www.colorado.edu/physics/phet/dev/html/beers-law-lab/1.3.0-dev.3/beers-law-lab_en.html?screenshot  ->  `-Beer's Law Lab- screenshot.png`
http://www.colorado.edu/physics/phet/dev/html/concentration/1.2.0-dev.6/concentration_en.html?screenshot -> `-Concentration- screenshot.png`

In my case, my browser is configured to saved downloads to $HOME/Downloads, which contains lots of files.  I went looking for the screenshot files in alphabetical order, and they weren't where I thought they would be.  It took me awhile to realize that the filenames all start with a hypen.

**Comment by jonathanolson** [comment]
Seems like it's mapping "out-of-range" characters (UTF8 embedding marks) to hyphens at the browser level.

I'll look into the best way to handle this (stripping of embedding marks perhaps?)

**Comment by pixelzoom** [comment]
Ah... You must mean the directional formatting (`<202a>` and `<202c>`) that surrounds the title.

**Comment by pixelzoom** [comment]
Does the file need to contain the sim name, or would something generic like `screenshot.png` or `phew-screenshot.png` be acceptable?

**Comment by pixelzoom** [comment]
Looks like escape characters aren't the only problem. Most operating systems have some chars that can't be used in filenames, and there's nothing to prevent those chars from being used in sim titles. For example, on Mac OS X, semicolon can be used in filenames, and "States Of Matter: Basics" screenshots are named `-States of Matter- Basics- screenshot.png`.

**Comment by samreid** [comment]
What about using the non-internationalized sim name from package.json "name"?  Those are already sanitized and standardized.

**Comment by pixelzoom** [comment]
I hadn't thought about i18n. Do screenshot filenames need to be internationalized? Does "screenshot" (as it appears in the filename) need to be a translatable joist string (its currently not)?

**Comment by jonathanolson** [comment]
> Do screenshot filenames need to be internationalized? Does "screenshot" (as it appears in the filename) need to be a translatable joist string (its currently not)?

Decisions probably pending on more feature design, since it's on hold. Should we disable the ability to add the query parameter flag?

Also, we can just use the internal sim name (states-of-matter-basics) if i18n is problematic here.

**Comment by pixelzoom** [comment]
> Should we disable the ability to add the query parameter flag?

Since KP just asked me to publish dev versions with this feature for demoing at a conference, I think not.

**Comment by pixelzoom** [comment]
> Should we disable the ability to add the query parameter flag?

There is aready a query parameter, `screenshot`. Is this what you mean?

**Comment by jonathanolson** [comment]
> There is aready a query parameter, screenshot. Is this what you mean?

I mean, is it a problem that we even provide the query parameter, instead of unconditionally having it disabled?

**Comment by pixelzoom** [comment]
> I mean, is it a problem that we even provide the query parameter, instead of unconditionally having it disabled?

Since KP asked me to publish some dev versions to demo the screenshots feature at a conference, we need some way to easily enable the feature.  Whether it's OK to publish sims to the PhET website with support for the "screenshot" feature is a question that should probably be directed to @ariel-phet.

**Comment by pixelzoom** [comment]
> I mean, is it a problem that we even provide the query parameter, instead of unconditionally having it disabled?

KP asked me (yesterday) to publish some dev versions to demo the screenshot feature at a conference, so we need some way to easily enable the feature.  Whether it's OK to publish sims to the PhET website with support for the "screenshot" query param is a question that should probably be directed to @ariel-phet.

**Comment by ariel-phet** [comment]
This issue appears like it can be closed
